### PR TITLE
init subscription support, test pass

### PR DESCRIPTION
**Pull request by myidsan**
subscription to check if the user is subscribed to Perf.
Check documentation for use.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -118,6 +119,20 @@ def get_accord_by_name(name):
     elif request.method == 'DELETE':
         return AccordView.delete_accord(name)
 
+@app.route('/subscription', methods=['POST', 'GET'])
+def subscription():
```
Comment (Mark-Jung) [inline app.py]: So for the subscription checking will be done by get_subscription_by_email, and I'll validate or reject the user based on the status code, correct?

Comment (myidsan) [reply app.py]: correct you can check by the status code returned from get_subscritption_by_email. Endpoint is /subscription/<string:email>.

**Comment by jameswhang** [comment]
Remind me what this is gonna be used for?

File: models/subscription.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+"""
+User
```
[the remark this answers is not in the document]
Comment (myidsan) [reply models/subscription.py]: This is not copy and paste. I started as making this model called user but changed to subscription - just forgot to change the upper desc my b.

File: models/subscription.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+from db import db
+from models.common import CommonModel
+
+class SubscriptionModel(db.Model, CommonModel):
+    __tablename__ = 'user'
+
+    first_name = db.Column(db.String(255))
+    last_name = db.Column(db.String(255))
+    email = db.Column(db.String(500), primary_key=True)
+    date_created = db.Column(db.DateTime, default=db.func.current_timestamp())
```
[the remark this answers is not in the document]
Comment (myidsan) [reply models/subscription.py]: We haven't discussed in-depth about what this will do. We will follow up on this.

[the remark this answers is not in the document]
Comment (myidsan) [reply models/subscription.py]: We haven't discussed in-depth about what this will do. We will follow up on this.

[the remark this answers is not in the document]
Comment (jameswhang) [reply models/subscription.py]: So you just made a PR on something you don't even know why it's there. Why bother do this?

[the remark this answers is not in the document]
Comment (myidsan) [reply models/subscription.py]: Let me rephrase: as far as we discussed, checking if the user was subscribed or not was the only feature.  We were not thinking hard enough about the overall features.

File: controllers/subscription.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        return "", 200, target_subscription
+
+    @classmethod
+    def cancel_subscription(cls, email):
+        if not SubscriptionModel.find_by_email(email):
+            return "A subscription with given email does not exists", 400, None
+
+        try:
+            target_subscription = SubscriptionModel.find_by_email(email)
+            target_subscription.delete_from_db()
```
[the remark this answers is not in the document]
Comment (Mark-Jung) [reply controllers/subscription.py]: @jameswhang  Because it can throw referential integrity errors?

[the remark this answers is not in the document]
Comment (jameswhang) [reply controllers/subscription.py]: Why are you commenting on a 2 month old PR..?

[the remark this answers is not in the document]
Comment (Mark-Jung) [reply controllers/subscription.py]: Nostalgia
